bootstrapvz/plugins/admin_user/tasks.py

```diff
@@ -42,13 +43,62 @@ def run(cls, info):
 
 
 class AdminUserCredentials(Task):
-	description = 'Modifying ec2-get-credentials to copy the ssh public key to the admin user'
+	description = 'Set up access credentials for the admin user'
 	phase = phases.system_modification
-	predecessors = [InstallInitScripts]
+	predecessors = [InstallInitScripts, CreateAdminUser]
 
 	@classmethod
 	def run(cls, info):
-		from bootstrapvz.common.tools import sed_i
-		getcreds_path = os.path.join(info.root, 'etc/init.d/ec2-get-credentials')
-		username = info.manifest.plugins['admin_user']['username']
-		sed_i(getcreds_path, 'username=\'root\'', 'username=\'{username}\''.format(username=username))
+                from bootstrapvz.common.exceptions import TaskError
+                from bootstrapvz.common.tools import sed_i
+                from bootstrapvz.common.tools import log_check_call
+                log = logging.getLogger(__name__)
+
+                if 'password' in info.manifest.plugins['admin_user']:
+                        if 'pubkey' in info.manifest.plugins['admin_user']:
+                                msg = 'The options password and pubkey are mutually exclusive'
+                                raise TaskError(msg)
+                        log_check_call(['chroot', info.root, 'chpasswd'],
+                                       info.manifest.plugins['admin_user']['username'] +
+                                       ':' + info.manifest.plugins['admin_user']['password'])
+                        return
+
+                getcreds_path = os.path.join(info.root, 'etc/init.d/ec2-get-credentials')
+                if 'pubkey' in info.manifest.plugins['admin_user']:
+                        import stat
+                        from shutil import copy
+                        full_path = info.manifest.plugins['admin_user']['pubkey']
+                        if not os.path.exists(full_path):
```

Comment:
Do this in `validate_manifest()` ([docs here](http://bootstrap-vz.readthedocs.org/en/master/developers/plugins.html)). This can be really annoying if it fails, you are almost done bootstrapping your image only to discover that you've mistyped a path :-(